src/jgikbase/idmapping/service/mapper_service.py

```diff
@@ -95,13 +95,13 @@ def _log_exception(err: Exception):
 
 
 def _format_error(
-    err: Exception, httpcode: int, errtype: ErrorType = None, errprefix: str = ""
+    err: Exception, httpcode: int, errtype: Optional[ErrorType] = None, errprefix: str = ""
 ):
     errjson = {
         "httpcode": httpcode,
         "httpstatus": responses[httpcode],
         "message": errprefix + str(err),
-        "callid": flask_req_global.req_id,
+        "callid": getattr(flask_req_global, 'req_id', None),
```

Comment:
Why is this necessary? There should always be a request ID. Not having one is an error and should cause a failure, not be silently ignored.

Reply:
It does have that attribute though, it's set here: https://github.com/jgi-kbase/IDMappingService/blob/9d9f01662c4b09ac873174b7119d62828965e116/src/jgikbase/idmapping/service/mapper_service.py#L217-L225

I would add an annotation to ignore the mypy complaint

Reply:
👍